### PR TITLE
feat(recommend): split recommend and ui-components repositories

### PR DESCRIPTION
**Pull request by shortcuts**
**Summary**

This PR removes everything related to the `ui-component-horizontal-sider` packages which will now be available here: https://github.com/algolia/ui-components.

**Result**

Only the packages related to Algolia Recommend UI components are on this repository.

### CODE REVIEW

**Comment by shortcuts** [comment]
Update: keeping this PR as a draft until the one for `ui-components` isn't fully finished

File: examples/demo/src/App.tsx [hunk 1]
```diff
@@ -26,19 +26,17 @@ const searchClient = algoliasearch(appId, apiKey);
...
 
 insights('init', { appId, apiKey });
+insights('setUserToken', 'user-token-1');
 
-function RecommendedItem({ item }) {
-  return <Hit hit={item} insights={insights} />;
-}
-
-function BundleItem({ item }) {
+function BundleItem({ item, onSelect }: BundleItemProps) {
```
Comment (francoischalifour) [inline examples/demo/src/App.tsx]: We can make this component generic.

**Review by francoischalifour — APPROVED** [review]
Great—feel free to merge this once the UI components repo is created from #50.

File: examples/demo/src/App.tsx (hunk 1 above)
Comment (shortcuts) [reply examples/demo/src/App.tsx]: Sure, done here: https://github.com/algolia/recommend/pull/48/commits/59023d00c034a53f8adfcd29f386a7181c790755

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**